### ISSUE TITLE
supabase.auth.linkIdentity does not return raw_user_meta_data

### ISSUE THREAD

**Issue by chanmathew** [supabase/auth#1708]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When calling `supabase.auth.linkIdentity` using the JS client, it does not return and store the `raw_user_meta_data` that you would normally get when using `supabase.auth.signInWithOAuth`:

![CleanShot 2024-08-04 at 16 06 01@2x](https://github.com/user-attachments/assets/1e386a30-f793-4167-b303-e6ad90a2b532)

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Sign in an anonymous user
2. Try to link identity with supabase.auth.linkIdentity
3. Check the raw_user_meta_data in auth.users

## Expected behavior

It should return the metadata just like an oauth sign in.

## Screenshots

When using supabase.auth.signInWithOAuth, it returns the raw_user_meta_data:
![CleanShot 2024-08-04 at 16 08 06@2x](https://github.com/user-attachments/assets/2dd462a3-d182-401d-b54e-645039a7a109)

When using supabase.auth.linkIdentity, it does not return it:
![CleanShot 2024-08-04 at 16 06 01@2x](https://github.com/user-attachments/assets/1e386a30-f793-4167-b303-e6ad90a2b532)

## System information

- Version of supabase-js: 2.44.2
- Version of Node.js: 20.15.1

## Additional context

It would be best if it can be returned as I want my postgres trigger to automatically add the full_name and avatar_url to my public.profiles table once the user's anonymous account has been converted to a permanent account once they've signed in.

**Comment by j4w8n-malynium** [comment]
How are you getting the screenshot output? I tried this with hosted Supabase, looked in the auth.users table in the dashboard, and the meta data was there for the user.

**Comment by j4w8n** [comment]
My apologies... commenting again, because my previous one was from the test account.

How are you getting the screenshot output? I tried this with hosted Supabase, looked in the auth.users table in the dashboard, and the user metadata was there for the user. I also console.log'd the returned `data` from the `linkIdentity` method, and it also showed the user metadata.

**Comment by chanmathew** [comment]
@j4w8n I'm getting it in the authentication section, and then looking for my user:
![CleanShot 2024-08-04 at 23 02 50@2x](https://github.com/user-attachments/assets/84a616a5-7b0d-4e73-b71a-782587548c3b)

Then I click on view user info:
![CleanShot 2024-08-04 at 23 03 06@2x](https://github.com/user-attachments/assets/ad5156ef-ee01-4b3c-b24b-35b3999c207d)

I've tried both local dev and in the cloud dashboard, seems like the raw_user_meta_data is empty in both.

**Comment by j4w8n** [comment]
I'm sorry. I misunderstood what you were saying. I now understand that you mean the metadata that comes back from the OAuth provider.

After converting the user, if I logout and log back in with OAuth, then the data is there, but it would be nice if that data came back during the conversion.

Based on what I see in the `data.url` of a conversion, it's authorizing directly to the OAuth provider:
```
'https://github.com/login/oauth/authorize?client_id=<redacted>&redirect_to=http%3A%2Flocalhost%3A5173%2Fself&redirect_uri=https%3A%2F%2F<redacted>.supabase.co%2Fauth%2Fv1%2Fcallback&response_type=code&scope=user%3Aemail&skip_http_redirect=true&state=<redacted>'
```
Versus an OAuth login `data.url`:
```
https://<redacted>.supabase.co/auth/v1/authorize?provider=github&redirect_to=http%3A%2F%2Flocalhost%3A5173%2Fauth%2Fcallback%3Fnext%3D%2Fapp&code_challenge=l3tDoN9qS2u2B3rLXGidmfDcQRga9h3tbhM-w_U21FU&code_challenge_method=s256
```

I tried to work around it by adding something like `?refresh=true` to the `redirectTo` option on `linkIdentity`, but the redirect never happens - I assume because it's authorizing directly to the provider, and in this case, I couldn't find any redirect url settings for GitHub, so it just sent me back to the app url set in the GitHub OAuth app.

**Comment by chanmathew** [comment]
@j4w8n Yeah that's right, the metadata doesn't come back during the conversion.

As for the redirectTo, my understanding is that it only redirects after the authentication, so I just use that to redirect to a specific page within my app.

**Comment by jonnylangefeld** [comment]
I just came here because I ran into the same issue.
It is true that if the user logs out and in again after a `supabase.auth.linkIdentity` the `raw_user_meta_data` is there, but it should be there even with just `linkIdentity`. 

Any fix or workaround would be appreciated.

**Comment by chanmathew** [comment]
Any updates on this from the Supabase team?

**Comment by DiegoSalazar27** [comment]
Same issue here

**Comment by abdullah-cod9** [comment]
Same issue here.

**Comment by PaulBarnesUK** [comment]
For anyone still running into this issue, I was able to workaround it by getting the metadata from the `identity_data` column on the `auth.identities` table - that does get populated as expected after linking.
